File: java/09_PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0) {
            return false;
        }

        long result = 0;
        int original = x;
        while(x != 0) {
            result = result * 10 + x % 10;
            x = x / 10;
        }

        return original == result;
    }
}

File: swift/09_PalindromeNumber.swift
// https://leetcode.com/problems/palindrome-number/
class Solution {
    func isPalindrome(_ x: Int) -> Bool {
        if (x < 0) {
            return false
        }
        var origin = x
        var result = 0
        while (origin != 0) {
            result = result * 10 + origin % 10
            origin = origin / 10
        }
        return x == result
    }
}
